bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java

```diff
@@ -269,12 +270,18 @@ private void executeEventCommands(List<Event> events, CommandTagType execTime) {
      * Migration for last_update-channel as this change is compatible to previous instances.
      */
     private void migrateLastUpdateChannel() {
-        Thing thing = getThing();
+        final Thing thing = getThing();
         if (thing.getChannel(CHANNEL_LAST_UPDATE) == null) {
             logger.debug("last_update channel is missing in this Thing. Adding it.");
-            ThingBuilder thingBuilder = editThing();
-            ChannelBuilder channelBuilder = ChannelBuilder.create(new ChannelUID(thing.getUID(), CHANNEL_LAST_UPDATE));
-            thingBuilder.withChannel(channelBuilder.withType(LAST_UPDATE_TYPE_UID).build());
+            final ThingBuilder thingBuilder = editThing();
```

Comment:
One minor improvement: You can call this method shortly before `withChannel(...)`.

Reply:
Yep. The ChannelBuilder can also be final. Done.